intake_geopandas/geopandas.py

```diff
@@ -70,28 +83,80 @@ def __init__(self, urlpath, bbox=None,
         self._bbox = bbox
         self._geopandas_kwargs = geopandas_kwargs or {}
         self._dataframe = None
+        self.storage_options = storage_options or {}
+
+        # warn if using fsspec caching and same_names not True for zip files
+        if 'cache' in self.urlpath and 'zip' in self.urlpath:
+            same_names = False  # default
+            # find different same_names setting
+            for c in ['blockcache', 'filecache', 'simplecache']:
+                if c in self.storage_options:
+                    if 'same_names' in self.storage_options[c]:
+                        same_names = self.storage_options[c]['same_names']
+            if not same_names:
+                warnings.warn(
+                    'Need same_names = True for local caching of `zip` files.'
+                )
 
         super().__init__(metadata=metadata)
 
     def _open_dataset(self):
         """
         Open dataset using geopandas and use pattern fields to set new columns.
         """
+        if 'cache' in self.urlpath:  # new caching
```

Comment:
What if the URL has `'cache'` in it? Doesn't this break for a whole class of valid URLs/paths?

Reply:
now checking for `cache::`

Reply:
Could also be "*cache://". Should maybe raise an issue on fsspec, so that we can tell whether a given URL/filesystem can return a local path.